lib/src/command/global_activate.dart

```diff
@@ -29,7 +31,8 @@ class GlobalActivateCommand extends PubCommand {
     argParser.addOption('git-path', help: 'Path of git package in repository');
 
     argParser.addOption('git-ref',
-        help: 'Git branch or commit to be retrieved');
+        help: 'Git branch, tag, or commit SHA to be retrieved. '
+            'When using a commit SHA, must be the full SHA (40 characters).');
```

Comment:
My initial fix was only able to handle full commit SHAs, but it turns out that partial commit SHAs now work as well with the current fix! So it's great that you looked over this because that made me test it once again with partial SHAs. I tested it locally with a remote git repository and added a test for this as well.